pkg/kubelet/dockertools/manager_test.go

```diff
@@ -115,6 +115,34 @@ func matchString(t *testing.T, pattern, str string) bool {
 	return match
 }
 
+func TestNewDockerVersion(t *testing.T) {
+	cases := []struct {
+		value string
+		out   string
+		err   bool
+	}{
+		{value: "1", err: true},
+		{value: "1.8", err: true},
+		{value: "1.8.1", out: "1.8.1"},
+		{value: "1.8.1.fc21", out: "1.8.1-fc21"},
+		{value: "1.8.1.fc21.other", out: "1.8.1-fc21.other"},
+		{value: "1.8.1-fc21.other", out: "1.8.1-fc21.other"},
```

Comment:
nit: @aveshagarwal's test case should be added here:
`{value: "1.8.1-beta.12", out: "1.8.1-beta.12"} `